component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java

```diff
@@ -1809,6 +1826,59 @@ public void onProcessor(final Class<?> type, final Processor processor) {
+                        final ParameterMeta umbrella = new ParameterMeta(new ParameterMeta.Source() {
+
+                            @Override
+                            public String name() {
+                                return "$configuration";
+                            }
+
+                            @Override
+                            public Class<?> declaringClass() {
+                                return Object.class;
+                            }
+                        }, Object.class, ParameterMeta.Type.OBJECT, "$configuration", "$configuration", new String[0],
+                                new ArrayList<>(), new ArrayList<>(), new HashMap<>(), true);
+                        parameterMetas.add(umbrella);
+                        return umbrella;
+                    });
+
+            final StreamingMaxRecordsParamBuilder parmBuilder = new StreamingMaxRecordsParamBuilder(root,
+                    type.getSimpleName(),
+                    org.talend.sdk.component.api.service.configuration.LocalConfiguration.class.cast(services.services
+                            .get(org.talend.sdk.component.api.service.configuration.LocalConfiguration.class)));
+            final ParameterMeta maxRecords = parmBuilder.newBulkParameter();
+            final ParameterMeta maxDuration = new StreamingMaxDurationMsParamBuilder(root, type.getSimpleName(),
+                    LocalConfiguration.class.cast(services.services.get(LocalConfiguration.class))).newBulkParameter();
+            final String layoutOptions = maxRecords.getName() + "|" + maxDuration.getName();
+            if (maxRecords != null) {
+                final String layoutType = parmBuilder.getLayoutType();
+                if (layoutType == null) {
+                    root.getMetadata().put("tcomp::ui::gridlayout::Advanced::value", layoutOptions);
+                    root
+                            .getMetadata()
+                            .put("tcomp::ui::gridlayout::Main::value",
+                                    root
+                                            .getNestedParameters()
+                                            .stream()
+                                            .map(ParameterMeta::getName)
+                                            .collect(joining("|")));
+                } else if (!root.getMetadata().containsKey(layoutType)) {
+                    root
+                            .getMetadata()
+                            .put(layoutType, layoutType.contains("gridlayout") ? layoutOptions : "true");
+                } else if (layoutType.contains("gridlayout")) {
+                    final String oldLayout = root.getMetadata().get(layoutType);
+                    root.getMetadata().put(layoutType, layoutOptions + "|" + oldLayout);
+                }
+            }
+            root.getNestedParameters().add(maxRecords);
```

Comment:
layoutType == null
maxRecord & maxDurations go to "Advanced"
getNested -> "Main"

and at the end we add maxRecords & maxDuration into the "getNested"
(frankly the idea goes away from me with those layouts)

Reply:
That's about default layout creation only. In all cases, maxR and maxD must be in nestedParameters.